core/src/test/scala/unit/kafka/server/ReplicaFetcherTierStateMachineTest.scala

```diff
@@ -121,4 +129,38 @@ class ReplicaFetcherTierStateMachineTest {
     assertTrue(fetcher.fetchState(partition).isEmpty)
     assertTrue(failedPartitions.contains(partition))
   }
+
+  @Test
+  def buildRemoteLogAuxStateShouldThrowErrorForUninitializedPartition(): Unit = {
+    val t1p0 = new TopicPartition("topic1", 0)
```

Comment:
nit: Rename it as `partition` as there are no other partition instances to differentiate with numbers.